packages/syft/tests/syft/core/tensor/smpc/smpc_tensor_test.py

```diff
@@ -132,3 +132,186 @@ def test_mpc_forward_methods(method_str: str, kwargs: TypeDict[str, Any]) -> Non
     expected = op(**kwargs)
 
     assert (res == expected).all()
+
+
+def test_repeat() -> None:
+    value = np.array([[1, 2], [3, 4]])
+
+    remote_value = clients[0].syft.core.tensor.tensor.Tensor(value)
+
+    mpc_tensor = MPCTensor(
+        parties=clients, secret=remote_value, shape=(2, 2), seed_shares=42
+    )
+
+    res = mpc_tensor.repeat(3).reconstruct()
+    exp_res = value.repeat(3)
+
+    assert (res == exp_res).all()
+
+
+def test_copy() -> None:
+    value = np.array([[1, 2], [3, 4]])
+
+    remote_value = clients[0].syft.core.tensor.tensor.Tensor(value)
+
+    mpc_tensor = MPCTensor(
+        parties=clients, secret=remote_value, shape=(2, 2), seed_shares=42
+    )
+
+    res = mpc_tensor.copy().reconstruct()
+    exp_res = mpc_tensor.reconstruct()
+    # we cannot check id for copy as the values are in different locations
+
+    assert (res == exp_res).all()
+
+
+def test_diagonal() -> None:
+    value = np.array([[0, 1], [2, 3]])
+
+    remote_value = clients[0].syft.core.tensor.tensor.Tensor(value)
+
+    mpc_tensor = MPCTensor(
+        parties=clients, secret=remote_value, shape=(2, 2), seed_shares=42
+    )
+
+    res = mpc_tensor.diagonal().reconstruct()
+    exp_res = value.diagonal()
+
+    assert (res == exp_res).all()
+
+
+def test_flatten() -> None:
+    value = np.array([[89, 12, 54], [412, 89, 42], [87, 32, 58]])
+
+    remote_value = clients[0].syft.core.tensor.tensor.Tensor(value)
+
+    mpc_tensor = MPCTensor(
+        parties=clients, secret=remote_value, shape=(3, 3), seed_shares=42
+    )
+
+    res = mpc_tensor.flatten().reconstruct()
+    exp_res = value.flatten()
+
+    assert (res == exp_res).all()
+
+
+def test_transpose() -> None:
+    value = np.array([[89, 12, 54], [412, 89, 42], [87, 32, 58]])
+
+    remote_value = clients[0].syft.core.tensor.tensor.Tensor(value)
+
+    mpc_tensor = MPCTensor(
+        parties=clients, secret=remote_value, shape=(3, 3), seed_shares=42
+    )
+
+    res = mpc_tensor.transpose().reconstruct()
+    exp_res = value.transpose()
+
+    assert (res == exp_res).all()
+
+
+@pytest.mark.xfail
+def test_partition() -> None:
+    value = np.array([12, 1, 12, 41, 8, 4, 7, 39, 27])
+
+    remote_value = clients[0].syft.core.tensor.tensor.Tensor(value)
+
+    mpc_tensor = MPCTensor(
+        parties=clients, secret=remote_value, shape=(9,), seed_shares=42
+    )
+
+    res = mpc_tensor.partition(5).reconstruct()
+    exp_res = value.partition(5)
+
+    assert (res == exp_res).all()
+
+
+@pytest.mark.xfail
+def test_resize() -> None:
```

Comment:
`ReSize`  tests fail due to refcheck in python, [Link](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.resize.html) ,will have to take additional look at it.

Reply:
Modified to copy